BUG: CustomBusinessMonthBegin and CustomBusinessMonthEnd type signatures are missing the calendar and weekmask keyword arguments

### Pandas version checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.


### Reproducible Example

```python
import numpy as np
import pandas as pd

cal = np.busdaycalendar()

pd.offsets.CustomBusinessMonthBegin(calendar=cal)
pd.offsets.CustomBusinessMonthEnd(calendar=cal)
```

### Issue Description

## Description

The type signatures for `CustomBusinessMonthBegin` and `CustomBusinessMonthEnd` are missing the `calendar` and `weekmask` keyword arguments, even though they are supported at runtime and documented.

By comparison, `CustomBusinessDay` correctly exposes the `calendar` parameter in its type signature.

As a result, valid code fails type checking with mypy.

## Reproducer

```python
import numpy as np
import pandas as pd

cal = np.busdaycalendar()

pd.offsets.CustomBusinessDay(calendar=cal)          # OK
pd.offsets.CustomBusinessMonthBegin(calendar=cal)   # mypy error
pd.offsets.CustomBusinessMonthEnd(calendar=cal)     # mypy error
```

mypy reports:

```text
error: Unexpected keyword argument "calendar" for "CustomBusinessMonthBegin"  [call-arg]
error: Unexpected keyword argument "calendar" for "CustomBusinessMonthEnd"  [call-arg]
```

## `reveal_type`

```python
from pandas.tseries.offsets import (
    CustomBusinessDay,
    CustomBusinessMonthBegin,
    CustomBusinessMonthEnd,
)

reveal_type(CustomBusinessDay)
reveal_type(CustomBusinessMonthBegin)
reveal_type(CustomBusinessMonthEnd)
```

Output:

```text
Revealed type is "def (
    n: int =,
    normalize: bool =,
    holidays: list[Any] =,
    calendar: pandas.tseries.holiday.AbstractHolidayCalendar | numpy._core.multiarray.busdaycalendar =
) -> pandas._libs.tslibs.offsets.CustomBusinessDay"

Revealed type is "def (
    n: int =,
    normalize: bool =,
    offset: datetime.timedelta =,
    holidays: list[Any] | None =
) -> pandas._libs.tslibs.offsets.CustomBusinessMonthBegin"

Revealed type is "def (
    n: int =,
    normalize: bool =,
    offset: datetime.timedelta =,
    holidays: list[Any] | None =
) -> pandas._libs.tslibs.offsets.CustomBusinessMonthEnd"
```

Notably, the month offset signatures are missing both `calendar` and `weekmask`.

## Expected behavior

The type signatures for `CustomBusinessMonthBegin` and `CustomBusinessMonthEnd` should match the documented/runtime API and accept the same customization parameters as the runtime implementation, including:

- `calendar`
- `weekmask`
- `holidays`

## Environment

- pandas 3.0.3
- mypy 2.1.0

### Documentation

The official documentation lists the `calendar` parameter:

- https://pandas.pydata.org/docs/reference/api/pandas.tseries.offsets.CustomBusinessMonthBegin.html
- https://pandas.pydata.org/docs/reference/api/pandas.tseries.offsets.CustomBusinessMonthEnd.html

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : 72f2fea91530b5abb3cf2100cb22d84e504695c0
python                : 3.14.6
python-bits           : 64
OS                    : Windows
OS-release            : 11
Version               : 10.0.26100
machine               : AMD64
processor             : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : English_United States.1252

pandas                : 3.0.3
numpy                 : 2.5.0
dateutil              : 2.9.0.post0
pip                   : 26.1.2
Cython                : None
sphinx                : None
IPython               : 9.15.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.15.0
bottleneck            : None
fastparquet           : None
fsspec                : 2026.6.0
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : 3.1.6
lxml.etree            : 6.1.1
matplotlib            : 3.11.0
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : 3.1.5
psycopg2              : None
pymysql               : None
pyarrow               : None
pyiceberg             : None
pyreadstat            : None
pytest                : None
python-calamine       : None
pytz                  : 2026.2
pyxlsb                : None
s3fs                  : 2026.6.0
scipy                 : 1.18.0
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : 2.0.2
xlsxwriter            : None
zstandard             : None
qtpy                  : None
pyqt5                 : None
</details>

--- a/pandas-stubs/_libs/tslibs/offsets.pyi
+++ b/pandas-stubs/_libs/tslibs/offsets.pyi
@@ -222,8 +222,10 @@ class _CustomBusinessMonth(SingleConstructorOffset):
         self,
         n: int = ...,
         normalize: bool = ...,
-        offset: timedelta = ...,
+        weekmask: str = ...,
         holidays: list[Any] | None = ...,
+        calendar: AbstractHolidayCalendar | np.busdaycalendar | None = ...,
+        offset: timedelta = ...,
     ) -> None: ...
 
 class CustomBusinessDay(BusinessDay):
@@ -246,8 +248,27 @@ class CustomBusinessHour(BusinessHour):
         holidays: list[Any] | None = ...,
     ) -> None: ...
 
-class CustomBusinessMonthEnd(_CustomBusinessMonth): ...
-class CustomBusinessMonthBegin(_CustomBusinessMonth): ...
+class CustomBusinessMonthEnd(_CustomBusinessMonth):
+    def __init__(
+        self,
+        n: int = ...,
+        normalize: bool = ...,
+        weekmask: str = ...,
+        holidays: list[Any] | None = ...,
+        calendar: AbstractHolidayCalendar | np.busdaycalendar | None = ...,
+        offset: timedelta = ...,
+    ) -> None: ...
+
+class CustomBusinessMonthBegin(_CustomBusinessMonth):
+    def __init__(
+        self,
+        n: int = ...,
+        normalize: bool = ...,
+        weekmask: str = ...,
+        holidays: list[Any] | None = ...,
+        calendar: AbstractHolidayCalendar | np.busdaycalendar | None = ...,
+        offset: timedelta = ...,
+    ) -> None: ...
 
 class DateOffset(RelativeDeltaOffset):
     def __init__(

--- a/pandas-stubs/_libs/tslibs/offsets.pyi
+++ b/pandas-stubs/_libs/tslibs/offsets.pyi
@@ -248,27 +248,8 @@ class CustomBusinessHour(BusinessHour):
         holidays: list[Any] | None = ...,
     ) -> None: ...
 
-class CustomBusinessMonthEnd(_CustomBusinessMonth):
-    def __init__(
-        self,
-        n: int = ...,
-        normalize: bool = ...,
-        weekmask: str = ...,
-        holidays: list[Any] | None = ...,
-        calendar: AbstractHolidayCalendar | np.busdaycalendar | None = ...,
-        offset: timedelta = ...,
-    ) -> None: ...
-
-class CustomBusinessMonthBegin(_CustomBusinessMonth):
-    def __init__(
-        self,
-        n: int = ...,
-        normalize: bool = ...,
-        weekmask: str = ...,
-        holidays: list[Any] | None = ...,
-        calendar: AbstractHolidayCalendar | np.busdaycalendar | None = ...,
-        offset: timedelta = ...,
-    ) -> None: ...
+class CustomBusinessMonthEnd(_CustomBusinessMonth): ...
+class CustomBusinessMonthBegin(_CustomBusinessMonth): ...
 
 class DateOffset(RelativeDeltaOffset):
     def __init__(

--- a/tests/test_holidays.py
+++ b/tests/test_holidays.py
@@ -5,6 +5,7 @@ from dateutil.relativedelta import MO
 import pandas as pd
 
 from tests import check
+import numpy as np
 
 from pandas.tseries.holiday import (
     AbstractHolidayCalendar,
@@ -45,3 +46,32 @@ def test_holiday_exclude_dates() -> None:
         exclude_dates=exclude,
     )
     check(assert_type(queens_jubilee_uk_spring_bank_holiday, Holiday), Holiday)
+
+def test_custom_business_month() -> None:
+
+    cal = np.busdaycalendar()
+
+    check(
+        assert_type(pd.offsets.CustomBusinessMonthBegin(calendar=cal),
+        pd.offsets.CustomBusinessMonthBegin),
+        pd.offsets.CustomBusinessMonthBegin,
+    )
+    check(
+        assert_type(pd.offsets.CustomBusinessMonthEnd(calendar=cal),
+        pd.offsets.CustomBusinessMonthEnd),
+        pd.offsets.CustomBusinessMonthEnd,
+    )
+    check(
+        assert_type(
+            pd.offsets.CustomBusinessMonthBegin(weekmask="Mon Tue Wed Thu Fri"),
+            pd.offsets.CustomBusinessMonthBegin,
+        ),
+        pd.offsets.CustomBusinessMonthBegin,
+    )
+    check(
+        assert_type(
+            pd.offsets.CustomBusinessMonthEnd(weekmask="Mon Tue Wed Thu Fri"),
+            pd.offsets.CustomBusinessMonthEnd,
+        ),
+        pd.offsets.CustomBusinessMonthEnd,
+    )

--- a/tests/test_holidays.py
+++ b/tests/test_holidays.py
@@ -2,10 +2,10 @@ from datetime import datetime
 from typing import assert_type
 
 from dateutil.relativedelta import MO
+import numpy as np
 import pandas as pd
 
 from tests import check
-import numpy as np
 
 from pandas.tseries.holiday import (
     AbstractHolidayCalendar,
@@ -47,18 +47,23 @@ def test_holiday_exclude_dates() -> None:
     )
     check(assert_type(queens_jubilee_uk_spring_bank_holiday, Holiday), Holiday)
 
+
 def test_custom_business_month() -> None:
 
     cal = np.busdaycalendar()
 
     check(
-        assert_type(pd.offsets.CustomBusinessMonthBegin(calendar=cal),
-        pd.offsets.CustomBusinessMonthBegin),
+        assert_type(
+            pd.offsets.CustomBusinessMonthBegin(calendar=cal),
+            pd.offsets.CustomBusinessMonthBegin,
+        ),
         pd.offsets.CustomBusinessMonthBegin,
     )
     check(
-        assert_type(pd.offsets.CustomBusinessMonthEnd(calendar=cal),
-        pd.offsets.CustomBusinessMonthEnd),
+        assert_type(
+            pd.offsets.CustomBusinessMonthEnd(calendar=cal),
+            pd.offsets.CustomBusinessMonthEnd,
+        ),
         pd.offsets.CustomBusinessMonthEnd,
     )
     check(

--- a/tests/test_holidays.py
+++ b/tests/test_holidays.py
@@ -1,4 +1,7 @@
-from datetime import datetime
+from datetime import (
+    datetime,
+    timedelta,
+)
 from typing import assert_type
 
 from dateutil.relativedelta import MO
@@ -76,6 +79,20 @@ def test_custom_business_month() -> None:
     check(
         assert_type(
             pd.offsets.CustomBusinessMonthEnd(weekmask="Mon Tue Wed Thu Fri"),
+            pd.offsets.CustomBusinessMonthEnd,
+        ),
+        pd.offsets.CustomBusinessMonthEnd,
+    )
+    check(
+        assert_type(
+            pd.offsets.CustomBusinessMonthBegin(offset=timedelta(hours=1)),
+            pd.offsets.CustomBusinessMonthBegin,
+        ),
+        pd.offsets.CustomBusinessMonthBegin,
+    )
+    check(
+        assert_type(
+            pd.offsets.CustomBusinessMonthEnd(offset=timedelta(hours=1)),
             pd.offsets.CustomBusinessMonthEnd,
         ),
         pd.offsets.CustomBusinessMonthEnd,

